File: Runners/JavaRunner/java-runner/src/main/java/com/mems/helpers/ITestWrapper.java
// ITestWrapper.java
package com.mems.helpers;

import com.mems.manifest.ManifestDto;

public interface ITestWrapper {
    /**
     * Generate full Java source (a single top-level class, ready to compile)
     */
    String generateSource(ManifestDto manifest, String languageCode, String userCode, String entrypointContainerClass, long defaultTimeoutMs)
            throws Exception;
}

File: Runners/KotlinRunner/src/main/kotlin/helpers/ITestWrapper.kt
package helpers

import manifest.ManifestDto

interface ITestWrapper {

    @Throws(Exception::class)
    fun generateSource(
        manifest: ManifestDto,
        languageCodeIn: String?,
        userCode: String?,
        entrypointContainerClassIn: String?,
        defaultTimeoutMs: Long
    ): String
}
